Update sites per plugin view for bulk management

cpapazoglou opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes https://github.com/Automattic/dotcom-forge/issues/9923

## Proposed Changes

* Adds a [new route](https://github.com/Automattic/wp-calypso/pull/97697/files#diff-7edd3c450e9fa353a0acd99fdf2b07d0c902ef5c88f419a3f9ddd3df242822f8) that loads the new plugins dashboard
* Adds a [condition](https://github.com/Automattic/wp-calypso/pull/97697/files#diff-33316ee02ed424c6862ab81c9792fd225a851305558dca401c9029a2dc2f94fb) so that Global Sidebar collapses as expected  
* Created a new standalone functional component [Plugins Dashboard](https://github.com/Automattic/wp-calypso/pull/97697/files#diff-8d1828ee17b44e66577d242e108e3f5216f9fbc9beb675ba587bab287bd7d9c4) that shows the plugins list and a flyout window when a plugin is clicked


## Known Issues

- [x] Styling per WHjPtm5mPBnJyiBYVHHnDT-fi-794_72587 and some mobile styling.
- [x] Bulk selecting plugins and activating / deactivating only works the second time attempted.
- [ ] The components used are from the `calypso/layout/multi-sites-dashboard` and use a lot of related classes. Need to come up with a better plan.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Visit `plugins/manage/sites` in this branch and `plugins/manage` in production
* Check that the dataview is rendered the same way
* Try bulk actions and make sure there are no regressions
* Click a plugin and check if the sites loaded are the same and if related actions work

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Plugins Management: replaces dialog with a new route

Commit:
Plugins Management: replaces dialog with a new route

Commit:
Plugins Management: replaces dialog with a new route

Commit:
Plugins Management: replaces dialog with a new route

Commit:
Plugins Management: add flyout panel

Commit:
Plugins Management: add flyout panel

Commit:
Removes unsused imports

Commit:
Removes unsused imports

Commit:
Makes the sidebar collapsible.

Commit:
Makes the sidebar collapsible.

Commit:
Temporary fixes flyout styling

Commit:
Temporary fixes flyout styling

Commit:
Sets plugins per view.

Commit:
Sets plugins per view.

Commit:
Further improvements

Commit:
Further improvements

Commit:
Fixes ts.

Commit:
Fixes ts.

Commit:
Fixes recordTracksEvents.

Commit:
Fixes recordTracksEvents.

Commit:
Adds sites list

Commit:
Adds sites list

Commit:
Fixes sites list

Commit:
Fixes sites list

Commit:
Fixes loader

Commit:
Fixes loader

Commit:
Fixes ts errors

Commit:
Fixes ts errors

Commit:
Reverts unused export

Commit:
Reverts unused export

gavande1 commented:
I have done some early testing for this PR. Everything seems to be working similarly to what is in production when you visit the `plugins/manage` route.

Apart from known issues, I found the following element out of place. I think it will be corrected while making the final layout changes. 
![CleanShot 2024-12-27 at 18 20 44@2x](https://github.com/user-attachments/assets/09c32090-5302-4d22-ba28-5a50291b944d)

Commit:
Adjusts headers copy

Commit:
Adjusts headers copy

gcsecsey committed:
pass selectedPlugins as an argument to action handlers

This change removes the need to use React state for tracking the selected plugins and avoids async side effects.

Commit:
Adjusts for rebase

Commit:
Removes stale styles

gcsecsey committed:
Merge branch 'update/sites-per-plugin-view-for-bulk-management' of github.com:Automattic/wp-calypso into update/sites-per-plugin-view-for-bulk-management

gcsecsey committed:
pass selectedPlugins as an argument to action handlers

This change removes the need to use React state for tracking the selected plugins and avoids async side effects.

Commit:
Fixes header styles.

Commit:
Properly gets site features

Commit:
Mobile fixes

cpapazoglou commented:
This is ready for a final review thanks to @gcsecsey and @gavande1 who helped.

Some notes:

- I have figured out that the designs have some extra functionality which can be tackled in another PR, after we add a sites dataview in the flyout pane
![image](https://github.com/user-attachments/assets/2f226277-5678-4c6e-a72e-700a34ca7bc2)
- I haven't styled the sites table since we aim to switch it to dataviews
- I think pushing for dataviews for all sites might be challenging for:
  - paid plugins
  - sites that need upgrade
  - sites that need Atomic transfer
 
Maybe we can start with a dataview only for sites that have the plugin installed.

gcsecsey reviewed (approved):
Changes LGTM and work as described. I tested the per-page actions and found no regressions. The mobile experience is also working nicer than the trunk version IMO.

The global nav collapse/expand and the panel animation also looks great! 🎉 
![CleanShot 2024-12-30 at 11 08 58](https://github.com/user-attachments/assets/cc0598f8-231b-4b6c-a6a3-d43e7f5d6271)

| Location | Trunk | This branch |
|--------|--------|--------|
| Plugin list | ![CleanShot 2024-12-30 at 11 03 04@2x](https://github.com/user-attachments/assets/fb1dcc49-3a0f-4347-a79d-ec7667a7b75c) | ![CleanShot 2024-12-30 at 11 01 30@2x](https://github.com/user-attachments/assets/099c3e21-011f-48b7-a138-4a0409d5fc6e) |
| Manage plugin | ![CleanShot 2024-12-30 at 11 03 10@2x](https://github.com/user-attachments/assets/71a76632-5e4e-4286-b86b-08532bd5bdea) | ![CleanShot 2024-12-30 at 11 02 03@2x](https://github.com/user-attachments/assets/f37c4996-d478-4932-96e9-89797b155e86) | 
| Plugin list mobile | ![CleanShot 2024-12-30 at 11 02 55@2x](https://github.com/user-attachments/assets/88f5f399-5e0e-4dc0-a36a-1d51fb1fdebb) | ![CleanShot 2024-12-30 at 11 02 27@2x](https://github.com/user-attachments/assets/8094e578-d0a6-4a83-9ae1-cbaaa5654e19) |
| Manage plugin mobile | ![CleanShot 2024-12-30 at 11 02 46@2x](https://github.com/user-attachments/assets/6c2d5428-02e6-47ae-85fe-a1d08e737aec) | ![CleanShot 2024-12-30 at 11 02 14@2x](https://github.com/user-attachments/assets/8b8ca4e3-96ea-4846-bbf8-d1a2bd8ac0d7) |

Commit:
Fixes types.

gcsecsey reviewed (changes requested):
I did another round of testing, checking for possible regressions on the A4A and Jetpack Manage dashboards too.

On A4A, I found no regressions, we don't show the plugin management on this dashboard yet.

| Trunk | PR |
|--------|--------|
| ![CleanShot 2024-12-31 at 11 49 40@2x](https://github.com/user-attachments/assets/a6ae95bb-aef2-4923-9f6a-6f9e3b343429) | ![CleanShot 2024-12-31 at 11 49 52@2x](https://github.com/user-attachments/assets/e3885729-0d6e-498a-a812-efed41d9a189) | 

For Jetpack Manage, I found a blocking issue, the new routes are not configured here yet, and when clicking a plugin, it shows an empty screen:

https://github.com/user-attachments/assets/672d9317-3bcd-4e19-be18-ba929824ebcc

I think this should be easy to resolve though.

gavande1 commented:
I noticed that on Jetpack Cloud, the header is center-aligned on the big screen. We should probably fix it as well.

![CleanShot 2024-12-31 at 19 37 51@2x](https://github.com/user-attachments/assets/6dde2154-ec9a-4c35-8a54-b3a4d10e1ffb)

gavande1 commented:
When visiting the individual plugin page, the page is not scrollable on Jetpack Cloud. 

https://github.com/user-attachments/assets/cbe1770b-9210-4f94-bcf8-1b868207648a

Commit:
Fixes Jetpack cloud

gcsecsey reviewed (approved):
Thanks for fixing the Jetpack Manage dashboard @cpapazoglou! 🙌 

I tested the changes again, and the fly-out panel functionality works as expected.
<img width="1850" alt="image" src="https://github.com/user-attachments/assets/28da7c87-a4b4-400a-b55f-e5bc1035e3e2" />

I also encountered the same styling issues as @gavande1 on the plugin page. Since this PR did not introduce them, I'd opt for creating individual issues to track them.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17089303

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @cpapazoglou, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Manage all your plugins in one place`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17089303).
